app/Community/Controllers/MessageController.php

```diff
@@ -6,28 +6,24 @@
 
 use App\Community\Actions\AddToMessageThreadAction;
 use App\Community\Actions\CreateMessageThreadAction;
+use App\Community\Requests\MessageRequest;
 use App\Http\Controller;
+use App\Models\Message;
 use App\Models\MessageThread;
 use App\Models\MessageThreadParticipant;
 use App\Models\User;
 use Illuminate\Http\RedirectResponse;
-use Illuminate\Http\Request;
-use Illuminate\Support\Arr;
-use Illuminate\Support\Facades\Validator;
 
 class MessageController extends Controller
 {
-    public function store(Request $request): RedirectResponse
+    public function store(MessageRequest $request): RedirectResponse
```

Comment:
Good to know. Going forward, should `Request` be wrapped for these form handling functions in controllers?

Reply:
I think so, yes, when applicable. Prior to refactoring a form to Livewire; any form that submits data to a controller's store or update method _can_ reference a specialized form request class that includes all the validation rules. There are arguments for not using form request classes but calling the validator explicitly in the controller method instead (as can be seen in legacy pages and the MessageController prior to this PR). Same goes for calling `authorize()` in a controller vs. applying a middleware on the route level. It's pretty much a matter of taste of the end of the day. As in, where do you expect to find information about how the request is handled, locally or on a lower level.
It is kinda related to writing single-purpose controllers with `__invoke()` vs. defining multiple CRUD methods per controller ("resource controller").

From what I can tell we will use Folio's file-base routing and more Livewire components. Both of these developments imply that authorization will be localized (Folio's middleware() and calling authorize() in controllers, yet, defining abilities in route middleware for legacy pages) and validation in request classes. However, it's up to you and @Jamiras to tell me which one you prefer - I'm familiar enough with all of those approaches to look for them when in doubt naturally. Otherwise I'd go with form request classes until further notice.